go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -927,3 +933,41 @@ func (bsnc *BaseSecondaryNetworkController) createVMSourcePodLSPOpsForLiveMigrat
 
 	return ops, nil
 }
+
+func (bsnc *BaseSecondaryNetworkController) configureVMPodLSPsAfterFailedLiveMigration(pod *corev1.Pod, nadName string) error {
+	kubevirtLiveMigrationStatus, err := kubevirt.DiscoverLiveMigrationStatus(bsnc.watchFactory, pod)
+	if err != nil {
+		return fmt.Errorf("failed to discover Live-migration status after pod termination: %w", err)
+	}
+	if kubevirtLiveMigrationStatus == nil ||
+		pod.Name != kubevirtLiveMigrationStatus.TargetPod.Name ||
+		kubevirtLiveMigrationStatus.State != kubevirt.LiveMigrationFailed {
+		return nil
+	}
+
+	// make sure sourcePod lsp is enabled if migration failed after DomainReady was set.
+	srcPod := kubevirtLiveMigrationStatus.SourcePod
+	sourcePodLspName := bsnc.GetLogicalPortName(srcPod, nadName)
+	sourcePodLsp := &nbdb.LogicalSwitchPort{Name: sourcePodLspName}
+	sourcePodLsp, err = libovsdbops.GetLogicalSwitchPort(bsnc.nbClient, sourcePodLsp)
+	if err != nil {
+		return fmt.Errorf("failed to fetch switch port for node %s: %w", sourcePodLspName, err)
+	}
+	sourcePodLsp.Enabled = ptr.To(true)
+
+	logicalSwitchName, err := bsnc.getExpectedSwitchName(srcPod)
+	if err != nil {
+		return fmt.Errorf("failed to fetch switch name for pod %s: %w", srcPod.Name, err)
+	}
+	ops, err := libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitchOps(bsnc.nbClient, nil, &nbdb.LogicalSwitch{Name: logicalSwitchName}, sourcePodLsp)
+	if err != nil {
+		return fmt.Errorf("failed updating logical switch port %+v on switch %s: %w", *sourcePodLsp, logicalSwitchName, err)
+	}
```

Comment:
If we do the encapsulation at previous comment we can replace this with
```golang
 ops, err = bsnc.setPodLogicalSwitchPortEnabledField(kubevirtLiveMigrationStatus.SourcePod, true)
```

Reply:
DONE